### DISCUSSION TITLE
Snack Details

### DISCUSSION THREAD

**Discussion by bvinson23** [nss-day-cohort-47/C47-Discussions#66] [Help Me] [answered]
Brandon Vinson

Have you searched the other help tickets? No other tickets on this topic

I'm trying to use expand on the fetch call to get the details of the snacks to appear on the dom.

List any error messages that appear:
SnackDetails.js:13 Uncaught (in promise) TypeError: Cannot read property 'name' of undefined

What have you tried? I put a console log in my SnackDetails.js after the function declaration to show me what was in the snackObject. It's giving me all the info I want for the single snack, but I can't figure out why it can''t read snackObject.type.name when that shows up as part of my object in the console.log.

https://github.com/nss-day-cohort-47/ldcc-bvinson23/blob/bv-detailsupdate/scripts/snacks/SnackDetails.js
https://github.com/nss-day-cohort-47/ldcc-bvinson23/blob/bv-detailsupdate/scripts/data/apiManager.js

**Comment by bvinson23** [comment, marked as answer]
Update: figured out my issue. Although the path I used to fetch worked in Postman, the browser didn't seem to. Instead apiURL/snacks/(a bunch of expanding and &'s) I added the ${snackId} right after the snacks/ then started the query.

**Reply by DavidEdwardDarden** [reply]
What you did was very helpful to me... but...What resources did you use to figure this out? (because my googling of json-servers relationship features was all over the place) Thanks for the post.

**Reply by bvinson23** [reply]
I wish I had some actual logic to tell you, but I was just trying things to see if they would work and got lucky.

**Reply by AutumnFoto** [reply]
thanks! worked for me
